glnx-fdio.c

```diff
@@ -247,7 +245,7 @@ glnx_link_tmpfile_at (GLnxTmpfile *tmpf,
   const gboolean replace = (mode == GLNX_LINK_TMPFILE_REPLACE);
   const gboolean ignore_eexist = (mode == GLNX_LINK_TMPFILE_NOREPLACE_IGNORE_EXIST);
 
-  g_return_val_if_fail (tmpf->src_dfd >= 0, FALSE);
+  g_return_val_if_fail (tmpf->fd >= 0, FALSE);
```

Comment:
It feels like we should just have both of them there, e.g.:

```c
g_return_val_if_fail (tmpf->src_dfd != -1, FALSE);
g_return_val_if_fail (tmpf->fd >= 0, FALSE);
```

?

That will still work with `AT_FDCWD`, right?
(I'd add the `src_dfd` check to `glnx_open_tmpfile_linkable_at()` as well).